staging/src/k8s.io/api/core/v1/types.go

```diff
@@ -533,6 +541,18 @@ type PersistentVolumeClaimSpec struct {
 	// (Alpha) Using the namespace field of dataSourceRef requires the CrossNamespaceVolumeDataSource feature gate to be enabled.
 	// +optional
 	DataSourceRef *TypedObjectReference `json:"dataSourceRef,omitempty" protobuf:"bytes,8,opt,name=dataSourceRef"`
+	// volumeAttributesClassName may be used to set the VolumeAttributesClass used by this claim.
+	// If specified, the CSI driver will create or update the volume with the attributes defined
+	// in the corresponding VolumeAttributesClass. This has a different purpose than storageClassName,
+	// it can be changed after the claim is created. An empty string value means that no VolumeAttributesClass
+	// will be applied to the claim but reset it to empty string is not allowed once it is set.
```

Comment:
What's the difference between:

- Setting VolumeAttributesClassName to an empty string
- Setting VolumeAttributesClassName to a VolumeAttributes resource where the Parameters are unset (since they are optional)
- Setting VolumeAttributesClassName to a VolumeAttributes resource where the Parameters are an empty map

?

Reply:
FYI https://github.com/kubernetes/kubernetes/pull/121104#discussion_r1375322044